openapi.yaml

```diff
@@ -171,6 +547,365 @@ components:
           schema:
             $ref: '#/components/schemas/ErrorResult'
   schemas:
+    BookId:
+      type: string
+      description: Book ID
+      example: '26'
+    BookType:
+      type: string
+      enum:
+        - book
+    BookAvailable:
+      type: string
+      nullable: false
+      description: Availability of book (true/false).
+      example: true
+    MemberId:
+      type: string
+      description: Member ID
+      example: '001'
+    MemberType:
+      type: string
+      enum:
+        - member
+    MembershipStatus:
+      type: string
+      nullable: false
+      description: Membership status of the member (active/suspended/expired).
+      example: active
+    BorrowId:
+      type: string
+      description: Borrow ID
+      example: 'bo123'
+    BorrowType:
+      type: string
+      enum:
+        - borrow
+    BorrowReturnDate:
+      type: string
+      format: date
+      nullable: true
+      description: Returned date of the book
+      example: null
+    BorrowStatus:
+      type: string
+      nullable: true
+      description: Current borrow status of the book (ongoing/returned/overdue)
+      example: ongoing
+    BookPostBody:
+      type: object
+      required:
+        - data
+      additionalProperties: false
+      properties:
+        data:
+          type: object
+          required:
+            - type
+            - attributes
+          additionalProperties: false
+          properties:
+            type:
+              $ref: '#/components/schemas/BookType'
+            attributes:
+              allOf:
+                - $ref: '#/components/schemas/BookPostRequired'
+                - $ref: '#/components/schemas/BookPostAttributes'
+    BookPostRequired:
+      type: object
+      required:
+        - title
+        - author
+        - publicationYear
+        - genre
+    BookPostAttributes:
+      type: object
+      additionalProperties: false
+      properties:
+        title:
+          type: string
+          nullable: false
+          description: Title of book
+          example: The Alchemist
+        author:
+          type: string
+          nullable: false
+          description: Author of book
+          example: Paulo Coelho
+        publicationYear:
+          type: integer
+          nullable: false
+          description: Publication year of book
+          pattern: ^\d{4}$
+          example: 1988
+        isbn:
+          type: string
+          nullable: true
+          description: International Standard Book Number (ISBN) for the book
+          example: 978-0061122415
+        genre:
+          type: string
+          nullable: false
+          description: Genre year of book
+          example: Fiction
+        description:
+          type: string
+          nullable: true
+          description: Description or summary of book
+          example: The Alchemist follows the journey of an Andalusian shepherd boy named Santiago...
```

Comment:
Should we allow max/min characters for this field?

Reply:
done